src/multiharp/device.rs

```diff
@@ -319,10 +421,40 @@ impl<T: MhlibWrapper> MH160 for MH160Device<T> {
     }
 }
 
+fn make_row_mask(channels_global: &[MH160ChannelId], rowidx: i32) -> i32 {
+    let row_start: i32 = rowidx * CHANNELS_PER_ROW + 1;
+    let row_end: i32 = row_start + CHANNELS_PER_ROW - 1;
+    let mut bits: u32 = 0;
+    for &global in channels_global {
+        let global_i32 = i32::from(global.0);
+        if (row_start..=row_end).contains(&global_i32) {
+            let local_i32 = global_i32 - row_start;
+            if let Ok(local) = u32::try_from(local_i32) {
+                bits |= 1u32 << local;
+            }
+        }
+    }
+    i32::try_from(bits).expect("row mask fits in i32")
```

Comment:
I think there's probably a conversion function from u32 that won't fail and allows use of all 32 bits, I'll have to check.